docs/en/sql-reference/statements/explain.md

```diff
@@ -10,7 +10,12 @@ Shows the execution plan of a statement.
 Syntax:
 
 ```sql
-EXPLAIN [AST | SYNTAX | PLAN | PIPELINE] [setting = value, ...] SELECT ... [FORMAT ...]
+EXPLAIN [AST | SYNTAX | PLAN | PIPELINE | TABLE OVERRIDE] [setting = value, ...]
+    [
+      SELECT ... |
+      tableFunction(...) [COLUMNS (...)] [ORDER BY ...] [PARTITION BY ...] [PRIMARY KEY] [SAMPLE BY ...] [TTL ...]
```

Comment:
btw why does it make sense to support, for example, `EXPLAIN TABLE OVERRIDE select * from mysql(...) PARTITION BY` if mysql table engine itself does not support partition by now? Same for TTL, SAMPLE BY, PRIMARY KEY.. (Looks like might make sense for materialized engine only, if so, then example with table function might seem confusing for users)

Reply:
Overriding storage parameters (which was my main reason for implementing table overrides) only makes sense for materialized engines of course. I understand that this syntax can be a bit confusing (mixing for example the `mysql()` table function up with the `MaterializedMySQL` database engine). It was just an implementation convenience, but could express it in some other way.

Some ideas:

(1)
```sql
EXPLAIN TABLE OVERRIDE
FOR TABLE 't1'
WITH ENGINE = MaterializedMySQL('host:port', 'db', 'user', 'pw')
PARTITION BY toYear(created)
```

This feels a bit heavy, but definitely more readable. Two more variants:

(2)
```sql
EXPLAIN TABLE OVERRIDE
t1 (PARTITION BY toYear(created))
ENGINE = MaterializedMySQL('host:port', 'db', 'user', 'pw')
```

This syntax is the closest to CREATE for overrides, which is nice.

(3)
```sql
EXPLAIN TABLE OVERRIDE
(PARTITION BY toYear(created))
FOR TABLE t1
WITH ENGINE = MaterializedMySQL('host:port', 'db', 'user', 'pw')
```

This is probably the one I like the least. (The parens around the actual override are needed to avoid parsing issues with TTLs.)

Using the ENGINE here lets users use the same external data source configuration as when you actually create the materialized database, which I think is a big plus.

What do you think?

Reply:
Oh, I see now how table function syntax here is nice and compact. So starting to lean towards it.

Reply:
Your comment made me think about it, and I do feel that being able to use external data source configurations is (or soon will be) important. But I guess that will be possible like this won't it?

```sql
EXPLAIN TABLE OVERRIDE mysql(myserver, table = 't1') PARTITION BY toYear(created)
```

Reply:
An argument for the `ENGINE = ...` syntax: by specifying which database engine you intend to use, we can implement different validation based on what the engine supports.  For example, I have some internal code for converting specific MySQL partitioning DDL to ClickHouse DDL, this would introduce a whole lot of very MySQL-specific validation.

Reply:
> But I guess that will be possible like this won't it?

Yes.
But which configuration parameters are present in storage definition, which are not present in table function and are at the same time relevant?

Reply:
> An argument for the ENGINE = ... syntax: 

ah, did not see this comment when answering above. Ok makes sense.

Reply:
So then we go for syntax (2). Thanks for your input!